grpc/README.md

```diff
@@ -1,5 +1,227 @@
+```java
+public ManagedChannel createChannel(String host, int port) {
+    return ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
+}
+```
+Please note that `usePlaintext()` should only be used for development and testing.
+When using in a production environment it should use a load balancer that terminates TLS.
+
+For local development of stargate, it will be:
+```java
+ManagedChannel channel = createChannel("localhost", 8090);
+```
+
+Next, we can generate the `StargateGrpc` stub. There are two ways of interacting with the gRPC API.
+
+The first one is synchronous (blocking):
+```java
+import io.stargate.grpc.StargateBearerToken;
+
+StargateGrpc.StargateBlockingStub blockingStub = StargateGrpc.newBlockingStub(channel)
+                                                    .withCallCredentials(new StargateBearerToken("token-value"))
+                                                    .withDeadlineAfter(5, TimeUnit.SECONDS);
+```
+
+The second one is async (non-blocking):
+```java
+StargateGrpc.StargateStub = = StargateGrpc.newStub(channel)
+                                .withCallCredentials(new StargateBearerToken("token-value"))
+                                .withDeadlineAfter(5, TimeUnit.SECONDS);
+```
+
+Please note, that we need setup the `CallCredentials`, using the `token-value` generated in the previous step.
+We also override the default [deadline](README.md#timeouts-deadlines).
+
+We will assume that all queries are executed within the existing keyspace `ks` and table `test`.
+The table definition is following:
+```cql
+CREATE TABLE IF NOT EXISTS test (k text, v int, PRIMARY KEY(k, v))
+``` 
+
+The Stargate gRPC API provides a way to execute two types of queries:
+- standard CQL queries
+- batch queries (contains N CQL queries)
+
+#### Standard Query
+
+We can start from inserting a record using gRPC stub:
+
+```java
+import io.stargate.proto.QueryOuterClass.Response;
+import io.stargate.proto.QueryOuterClass;
+
+Response response =
+        blockingStub.executeQuery(
+                        QueryOuterClass.Query
+                                    .newBuilder().setCql("INSERT INTO ks.test (k, v) VALUES ('a', 1)").build());
+```
+This will build and execute a single query. 
+
+
+Next, we can retrieve the inserted record(s):
+```java
+Response response = stub.executeQuery(QueryOuterClass.Query.newBuilder().setCql("SELECT k, v FROM ks.test").build());
+
+```
+
+If we print out the result set, it will have the following structure:
+```yaml
+result_set {
+  data {
+    type_url: "type.googleapis.com/stargate.ResultSet"
+    value: "some_binary_data"
+  }
+}
+```    
+The value contains the binary data, that we can deserialize. 
+Firstly, we need to unwrap the `ResultSet`:
+```java
+ResultSet rs = response.getResultSet().getData().unpack(QueryOuterClass.ResultSet.class);
+
+rs.getRows(0).getValues(0).getString(); // it will return value for k = "a"
```

Comment:
what about if we replace this with the example of get the row list and iterating over, some kind of example that is not so bounded on the type of the query, would this be possible? starter programmers could like this better I guess..

Reply:
done